viewing_party/party.py

```diff
@@ -1,23 +1,146 @@
-
+def get_unique_watched(user_data):
+    friends_movie_list = list()
+    unique_movies = list()
+    if len(user_data["watched"]) == 0:
+        unique_movies = []
+        return unique_movies
+    else:
+        for i in range(len(user_data["friends"])):
+            for j in range(len(user_data["friends"][i]["watched"])):
+                friends_movie_list.append(user_data["friends"][i]["watched"][j]["title"])
+        friends_movie_set = set(friends_movie_list)
+        for k in range(len(user_data["watched"])):
+            if user_data["watched"][k]["title"] not in friends_movie_set:
+                unique_movies.append(user_data["watched"][k])
+        return unique_movies
+    
+def get_friends_unique_watched(user_data):
+    friends_movie_list = list()
+    friends_unique_movies = list()
+    if len(user_data["watched"]) == 0:
+        for i in range(len(user_data["friends"])):
+            for j in range(len(user_data["friends"][i]["watched"])):
+                friends_movie_list.append(user_data["friends"][i]["watched"][j])
+        return friends_unique_movies
+    else:
+        for i in range(len(user_data["friends"])):
+            for j in range(len(user_data["friends"][i]["watched"])):
+                friends_movie_list.append(user_data["friends"][i]["watched"][j])
+        for k in range(len(friends_movie_list)):
+            if (friends_movie_list[k] not in user_data["watched"]) and ((friends_movie_list[k] not in friends_unique_movies)):
+                friends_unique_movies.append(friends_movie_list[k])
+        return friends_unique_movies#_set
         
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
-
+def get_available_recs(user_data):
+    list_of_recommended_movies = list()
+    unique_friends_movie = get_friends_unique_watched(user_data)
+    if not user_data["watched"] or not unique_friends_movie or not user_data["subscriptions"]:
+        return list_of_recommended_movies
+    else:
+        for i in range(len(unique_friends_movie)):
+            if unique_friends_movie[i]["host"] in user_data["subscriptions"]:
+                list_of_recommended_movies.append(unique_friends_movie[i])
+        if not list_of_recommended_movies:
+            return list_of_recommended_movies
+        else:
+            return list_of_recommended_movies
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
+def get_new_rec_by_genre(user_data):
+    recommended_movies = list()
+    list_of_genre = list()
+    if not user_data["watched"]:
+        return recommended_movies
+    else:
+        for movie in user_data["watched"]:
+            list_of_genre.append(movie["genre"])
+        most_ofen_genre = statistics.mode(list_of_genre)
+        list_of_movies = get_friends_unique_watched(user_data)
+        if not list_of_movies or not list_of_genre:

```

Comment:
We could forgo this if-check without impacting the outcome of the function. If `list_of_movies` is empty, then we'd never enter the loop on line 129, we don't need extra checks to prevent it. 
``` python
list_of_movies = get_friends_unique_watched(user_data)

for i in range(len(list_of_movies)):
    if list_of_movies[i]["genre"] == most_ofen_genre:
        recommended_movies.append(list_of_movies[i])

return recommended_movies
```